src/goto-harness/function_call_harness_generator.cpp

```diff
@@ -48,6 +48,9 @@ struct function_call_harness_generatort::implt
   std::set<irep_idt> function_parameters_to_treat_as_arrays;
   std::set<irep_idt> function_arguments_to_treat_as_arrays;
 
+  std::set<irep_idt> function_parameters_to_treat_as_cstrings;
```

Comment:
FWIW I think we need to start more cleanly separating between what we get for command-line options and what things we generate as intermediates based on that.